en/middleware/uorb.md

```diff
@@ -141,7 +141,40 @@ Make sure not to mix `orb_advertise_multi` and `orb_advertise` for the same topi
 
 The full API is documented in [platforms/common/uORB/uORBManager.hpp](https://github.com/PX4/PX4-Autopilot/blob/main/platforms/common/uORB/uORBManager.hpp).
 
+## Message Versioning
 
+<Badge type="tip" text="main (PX4 v1.16+)" />
+
+Optional message versioning was introduced in PX4 v1.16 (main) to make it easier to maintain compatibility between PX4 and ROS 2 versions.
+Versioned messages are designed to remain more stable over time compared to their non-versioned counterparts, as they are intended to be used across multiple releases of PX4 and external systems, ensuring greater compatibility over longer periods.
+
+Versioned messages include an additional field `uint32 MESSAGE_VERSION = x`, where `x` corresponds to the current version of the message.
+
+Versioned and non-versioned messages are separated in the file system:
+
+- Non-versioned topic message files and service message files remain in the [`msg/`](https://github.com/PX4/PX4-Autopilot/tree/main/msg) and [`srv/`](https://github.com/PX4/PX4-Autopilot/tree/main/srv) directories, respectively.
```

Comment:
I've subedited this a little, mainly "to my taste".

It is still a problem for me that this is the first and only mention of "topic message files and service message files" and that there is no definition of what these are for, or where they are from. If you're from ROS-land perhaps you know, but I can assure you that many readers will not.

Can you create a section in this topic which outlines the differences/similarities, and when you should create one or the other. Can be a different PR.